bcda/client/bluebutton.go

```diff
@@ -73,34 +73,34 @@ func NewBlueButtonClient() (*BlueButtonClient, error) {
 	return &BlueButtonClient{*client}, nil
 }
 
-func (bbc *BlueButtonClient) GetPatientData(patientID string) (string, error) {
+func GetPatientData(patientID string, bbc APIClient) (string, error) {
```

Comment:
Does this have to be a parameter instead of keeping the `BlueButtonClient` receiver? I thought that types implementing the interface could have additional functions, so `BlueButtonClient` would have to have `getData()` from `APIClient` but could have these other functions as well. (As a Go newbie I could be totally wrong here.)

Reply:
Interfaces can't have methods so unless I want to rewrite each of the GetPatientData, GetCoverageData .... methods for my mock client I have to pass it in.  I may also just be doing inheritance incorrectly in Go though.

Reply:
Maybe a few months from now when we're more Go-knowledgable we'll look back at this and say, "Oh, we totally could've used ____!" 😄 But this makes sense for being able to use the mock client. 👍

Reply:
probably.  I look forward to that day.